tests/infra/consortium.py

```diff
@@ -322,11 +317,17 @@ def trust_node(
             timeout=timeout,
         )
 
-        if not self._check_node_exists(
-            remote_node, node_id, NodeStatus.TRUSTED
-        ) and not self._check_node_exists(remote_node, node_id, NodeStatus.LEARNER):
+        if self.reconfiguration_type == "2tx":
+            # Note: the order of these checks is important
+            if not self._check_node_exists(
+                remote_node, node_id, NodeStatus.LEARNER
+            ) and not self._check_node_exists(remote_node, node_id, NodeStatus.TRUSTED):
+                raise ValueError(
+                    f"Node {node_id} does not exist in state {NodeStatus.TRUSTED} or {NodeStatus.LEARNER}"
+                )
+        elif not self._check_node_exists(remote_node, node_id, NodeStatus.TRUSTED):
             raise ValueError(
-                f"Node {node_id} does not exist in state {NodeStatus.TRUSTED} or {NodeStatus.LEARNER}"
+                f"Node {node_id} does not exist in state {NodeStatus.TRUSTED}"
             )
```

Comment:
AIUI, the test would pass if the node is fast enough to transition from `LEARNER` to `TRUSTED` before we check its state in the store. Wouldn't it be equivalent to `self.wait_for_node_to_exist_in_store(remote_node, node_to_retire.node_id, timeout, NodeStatus.TRUSTED)` for both 1tx and 2tx reconfiguration?

Reply:
Yes, I think that's right, I just didn't want to run the waiting loop here as it's fine if the node is still a learner.